spikeinterface/core/numpyextractors.py

```diff
@@ -287,3 +288,138 @@ def get_event_times(self, channel_id, start_time, end_time):
+                              snippet_len=snippets_list[0].shape[1], channel_ids=channel_ids,
+                              dtype=dtype)
+
+        self.is_dumpable = False
+
+        for snippets, spikesframes in zip(snippets_list, spikesframes_list):
+            snp_segment = NumpySnippetsSegment(snippets, spikesframes)
+            self.add_snippets_segment(snp_segment)
+
+        self._kwargs = {'snippets_list': snippets_list,
+                        'spikesframes_list': spikesframes_list,
+                        'nbefore': nbefore,
+                        'sampling_frequency': sampling_frequency,
+                        'channel_ids': channel_ids
+                        }
+
+
+class NumpySnippetsSegment(BaseSnippetsSegment):
+    def __init__(self, snippets, spikesframes):
+        BaseSnippetsSegment.__init__(self)
+        self._snippets = snippets
+        self._spikestimes = spikesframes
+
+    def get_snippets(self,
+                     indices,
+                     channel_indices: Union[List, None] = None,
+                     ) -> np.ndarray:
+        """
+        Return the snippets, optionally for a subset of samples and/or channels
+
+        Parameters
+        ----------
+        indexes: (Union[int, None], optional)
+            start sample index, or zero if None. Defaults to None.
+        end_frame: (Union[int, None], optional)
+            end_sample, or number of samples if None. Defaults to None.
+        channel_indices: (Union[List, None], optional)
+            Indices of channels to return, or all channels if None. Defaults to None.
+
+        Returns
+        -------
+        snippets: np.ndarray
+            Array of snippets, num_snippets x num_samples x num_channels
+        """
+        if indices is None:
+            return self._snippets[:,:,channel_indices]
+        return self._snippets[indices,:,channel_indices]
+
+    def get_num_snippets(self):
+        return self._spikestimes.shape[0]
+
+    def frames_to_indices(self,
+                          start_frame: Union[int, None] = None,
+                          end_frame: Union[int, None] = None):
+        """
+        Return the slice of snippets
+
+        Parameters
+        ----------
+        start_frame: (Union[int, None], optional)
+            start sample index, or zero if None. Defaults to None.
+        end_frame: (Union[int, None], optional)
+            end_sample, or number of samples if None. Defaults to None.
+
+        Returns
+        -------
+        snippets: slice
+            slice of selected snippets
+        """
+        # must be implemented in subclass
+        if start_frame is None:
+            init = 0
+        else:
+            init = np.searchsorted(self._spikestimes, start_frame, side='left')
+        if end_frame is None:
+            endi = self._spikestimes.shape[0]
+        else:
+            endi = np.searchsorted(self._spikestimes, end_frame, side='left')
+        return slice(init, endi, 1)
```

Comment:
This should be in BaseSnippetsSegment no ? because it will always be like this no ?